lib/ansible/plugins/connection/__init__.py

```diff
@@ -88,6 +89,16 @@ def __init__(self, play_context, new_stdin, shell=None, *args, **kwargs):
 
         self.become = None
 
+    @property
+    def _new_stdin(self):
+        display.deprecated(
```

Comment:
also, if we want to, we could keep this working with a `BytesIO` stdin simulator, but I'm kinda leaning toward "not worth the trouble"

Reply:
I think it would be an interesting exercise/fun but probably need more time. I could remove the deprecation if we want to do that.

Reply:
IIRC the only other consumer of that is `synchronize`, which as you know "there be dragons" :laughing: :dragon:

Reply:
For simplicity, I've decided not to do this - `synchronize` uses it to instantiate a new local connection (which is now optional), so the plugin will just be able to delete it.